invenio_communities/records/records/systemfields/communities/manager.py

```diff
@@ -0,0 +1,205 @@
+
+        # Unset default if needed
+        if self._default_id == community_id:
+            self._default_id = None
+
+    def clear(self):
+        """Clear all communities from the record."""
+        # Remove all associations
+        res = self._m2m_model_cls.query.filter_by(
+            record_id=self._record_id).delete()
+        self._communities_ids = set()
+        self._default_id = None
+        self._communities_cache = {}
+
+    def refresh(self):
+        """Refresh from the database M2M table."""
+        # Retrieve from M2M table
+        ids = db.session.query(
+            self._m2m_model_cls.community_id
+        ).filter(
+            self._m2m_model_cls.record_id == self._record_id
+        ).all()
+
+        # Set internal list
+        self._communities_ids = set([str(x[0]) for x in ids])
+
+        # Unset default if no longer available
+        if self._default_id and self._default_id not in self._communities_ids:
+            self._default_id = None
+
+    def __len__(self):
+        """Get number of communities"""
+        return len(self._communities_ids)
+
+    def __contains__(self, community_or_id):
+        """Check record is in community."""
+        id_ = self._to_id(community_or_id)
+        return id_ in self._communities_ids
+
+    def __iter__(self):
+        """Iterate over a communities."""
+        # Determine community ids not already cached.
+        nocache_ids = self._communities_ids - set(
+            self._communities_cache.keys())
+
+        # Fetch and cache missing community records
+        if nocache_ids:
+            communities = Community.get_records(nocache_ids)
+            for c in communities:
+                self._communities_cache[str(c.id)] = c
+
+        # Iterate (sort by identifier to ensure consistent results)
+        return (
+            self._communities_cache[c]
+            for c in sorted(self._communities_ids)
+        )
+
+    @property
+    def ids(self):
+        """Get communities ids."""
+        return sorted(self._communities_ids)
+
+    @property
+    def default(self):
+        """Get the default community."""
+        if self._default_id:
+            return self._lookup_community(self._default_id)
+        return None
+
+    @default.setter
+    def default(self, community_or_id):
+        """Set the default community.
+
+        Note, the community must already have been added to the community. If
+        not, then use ``.add(community, default=True)`` instead.
+        """
+        id_ = self._to_id(community_or_id)
+        if id_ not in self._communities_ids:
+            raise AttributeError(
+                'Cannot set community as the default. '
+                'The record has not been added to the community.'
+            )
+        self._default_id = id_
+
+    @default.deleter
+    def default(self):
+        self._default_id = None
+
+    # Persist relationships in record (denormalize the M2M table).
```

Comment:
In the end I went for denormalising the entire structure into the parent record so looks like this (originally I was only going to add ``"default"``:

```
{
    "default": "<uuid>",
    "ids": ["<uuid>", "<uuid>", ...],
}
```